core/src/main/kotlin/com/alecstrong/sql/psi/core/psi/mixins/ResultColumnMixin.kt

```diff
@@ -13,10 +14,14 @@ internal abstract class ResultColumnMixin(
 ) : SqlCompositeElementImpl(node),
     SqlResultColumn {
   private val queryExposed: Collection<QueryResult> by ModifiableFileLazy(containingFile) lazy@{
-    val parent = parent as? SelectStmtMixin ?: return@lazy emptyList<QueryResult>()
+    val fromQuery = when (val parent = parent) {
+      is SelectStmtMixin -> parent.fromQuery()
+      is PostgreSqlReturningClause -> emptyList()
```

Comment:
drop this case and just have the `else`? reason I'm thinking is if we keep hard references to dialects out of these base types it'll be easier to eventually let them be separate repos/artifacts

Reply:
The issue is is that the `else` is a guard clause, whereas the `is PostgreSqlReturningClause` check isn't. I'll have a think of how to get round this and remove the explicit dependency on PostgreSQL.

Reply:
is it ever valid to do something like

```
INSERT INTO blah
VALUES ?, ?
RETURNING blah.*;
```

or 

```
INSERT INTO blah
VALUES ?, ?
RETURNING *;
```

Reason I ask is that `fromQuery` only gets used for those two cases, and so maybe this code just needs to be refactored so that `expr?.let` happens first, and then if the parent isn't a select clause we can use the guard clause

Reply:
Yep, both of them are valid. Maybe a dumb question, but I assume that the `?` should be wrapped in parentheses? Otherwise, I don't believe it would be valid PostgreSQL.

```sql
INSERT INTO blah
VALUES (?), (?)
RETURNING blah.*;
```

or

```sql
INSERT INTO blah
VALUES (?), (?)
RETURNING *;
```

Reply:
yea it should be wrapped

alright and in both those cases it returns the entire projection of the things inserted?

i think if thats the case the proper way to handle this would be to have some new interface which exposes `fromQuery`, and then `PostgresReturningStmt` would have to implement that and expose the table projection there. I don't know if that's worth doing in this PR so maybe just leave a TODO or file an issue to track it for later

Reply:
Just pulled out into a separate issue https://github.com/AlecStrong/sql-psi/issues/173.